main.cpp

```diff
@@ -19,13 +19,23 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+	auto l_brace = cstmtptr->getLParenLoc();
+	auto r_brace = cstmtptr->getRParenLoc();
+	auto end_of_var = Lexer::getLocForEndOfToken(cstmtptr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(), 0, *Result.SourceManager, LangOptions());
+	auto start_of_var = cstmtptr->getSubExprAsWritten()->getBeginLoc();

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)`.

Reply:
getRParenLoc().getLocWithOffset(1) returns number of elem in brace. getSubExprAsWritten()->getBeginLoc() gets the begin location of the variable. The difference is that in the first case we can find out the location of the element we need

Reply:
`getRParenLoc().getLocWithOffset(1) returns number of elem in brace` - this is not correct.

Reply:
getSubExprAsWritten()->getBeginLoc() points to the beginning of the name of the variable that we're trying to cast
getRParenLoc().getLocWithOffset(1) points to the character that's right after the closing parenthesis of the type name to which we're trying to cast to
Which means if we use it, the spaces after the closing parenthesis will transfer with the name of the variable